include/libdnf/utils/utils.hpp

```diff
@@ -125,42 +135,110 @@ inline GoalProblem operator&(GoalProblem lhs, GoalProblem rhs) {
         static_cast<std::underlying_type<GoalProblem>::type>(rhs));
 }
 
-inline bool GoalJobSettings::get_strict(const libdnf::ConfigMain & cfg_main) const {
+inline bool GoalJobSettings::get_strict(const libdnf::ConfigMain & cfg_main,  bool set_used_value) {
     switch (strict) {
-        case GoalSetting::AUTO:
-            return cfg_main.strict().get_value();
+        case GoalSetting::AUTO: {
+            bool strict = cfg_main.strict().get_value();
+            if (set_used_value) {
+                if (used_strict != GoalUsedSetting::UNUSED) {
+                    throw LogicError("Used value for 'strict' already set");
```

Comment:
This concept seems to be a bit confusing from a user perspective.
You're calling getter `get_strict()` with a `set_used_value` argument and the setter eventually complains about setting something.

Reply:
I will rename the method to get_and_store_value. Is it better?